op-dispute-mon/mon/claims.go

```diff
@@ -15,59 +16,81 @@ type RClock interface {
 
 type ClaimMetrics interface {
 	RecordClaims(status metrics.ClaimStatus, count int)
-	RecordUnexpectedClaimResolution(address common.Address, count int)
+	RecordHonestActorClaims(address common.Address, data *metrics.HonestActorData)
 }
 
 type ClaimMonitor struct {
 	logger       log.Logger
 	clock        RClock
-	honestActors []common.Address
+	honestActors map[common.Address]bool // Map for efficient lookup
 	metrics      ClaimMetrics
 }
 
 func NewClaimMonitor(logger log.Logger, clock RClock, honestActors []common.Address, metrics ClaimMetrics) *ClaimMonitor {
-	return &ClaimMonitor{logger, clock, honestActors, metrics}
+	actors := make(map[common.Address]bool)
+	for _, actor := range honestActors {
+		actors[actor] = true
+	}
+	return &ClaimMonitor{logger, clock, actors, metrics}
 }
 
 func (c *ClaimMonitor) CheckClaims(games []*types.EnrichedGameData) {
 	claimStatus := make(map[metrics.ClaimStatus]int)
-	unexpected := make(map[common.Address]int)
+	honest := make(map[common.Address]*metrics.HonestActorData)
+	for actor := range c.honestActors {
+		honest[actor] = &metrics.HonestActorData{
+			PendingBonds: big.NewInt(0),
+			LostBonds:    big.NewInt(0),
+			WonBonds:     big.NewInt(0),
+		}
+	}
 	for _, game := range games {
-		c.checkGameClaims(game, claimStatus, unexpected)
+		c.checkGameClaims(game, claimStatus, honest)
 	}
 	for status, count := range claimStatus {
 		c.metrics.RecordClaims(status, count)
 	}
-	for address, count := range unexpected {
-		c.metrics.RecordUnexpectedClaimResolution(address, count)
+	for actor := range c.honestActors {
+		c.metrics.RecordHonestActorClaims(actor, honest[actor])
 	}
 }
 
-func (c *ClaimMonitor) checkResolvedAgainstHonestActor(proxy common.Address, claim *types.EnrichedClaim, unexpected map[common.Address]int) {
-	for _, actor := range c.honestActors {
-		if claim.Claimant == actor && claim.CounteredBy != (common.Address{}) {
-			unexpected[actor]++
+func (c *ClaimMonitor) checkUpdateHonestActorStats(proxy common.Address, claim *types.EnrichedClaim, honest map[common.Address]*metrics.HonestActorData) {
+	if !claim.Resolved {
+		if c.honestActors[claim.Claimant] {
+			honest[claim.Claimant].PendingClaimCount++
+			honest[claim.Claimant].PendingBonds = new(big.Int).Add(honest[claim.Claimant].PendingBonds, claim.Bond)
+		}
+		return
+	}
+	if c.honestActors[claim.Claimant] {
+		actor := claim.Claimant
+		if claim.CounteredBy != (common.Address{}) {
+			honest[actor].InvalidClaimCount++
+			honest[actor].LostBonds = new(big.Int).Add(honest[actor].LostBonds, claim.Bond)
 			c.logger.Error("Claim resolved against honest actor", "game", proxy, "honest_actor", actor, "countered_by", claim.CounteredBy, "claim_contract_index", claim.ContractIndex)
-			break
+		} else {
+			honest[actor].ValidClaimCount++
+			// Note that we don't count refunded bonds as won
```

Comment:
I'm not entirely convinced by this. For example, if an invalid output root is submitted, an honest actor challenges it, and no one counters it through resolution, this metric will fail to record that any bonds were won. But in that case shouldn't we show that an honest actor won a bond?

Reply:
The honest actor would win the bond for the root claim in that case.

I initially included refunded bonds and it just winds up as an ever increasing metric where it's impossible to see when the challenger takes someone else's bond because it gets lost in the amount of their own bonds they get back.  If we had total bonds posted and total bonds awarded as separate metrics you could extract it out again but I think the interesting metric is how much profit you've made so worth just capturing that to start with. Total bonds posted isn't a particularly useful metric because it double counts a lot of funds as they are posted and refunded - the total in-progress bonds is a much more useful view of how much you've got on the line.

Reply:
That makes sense to me, can always add that metric later if we really feel the need, but I'm thinking you're correct in that it won't be as useful.